packages/extension-horizontal-rule/src/horizontal-rule.ts

```diff
@@ -86,7 +93,7 @@ export const HorizontalRule = Node.create<HorizontalRuleOptions>({
                     }
                   } else {
                     // add node after horizontal rule if it’s the end of the document
-                    const node = $to.parent.type.contentMatch.defaultType?.create()
+                    const node = state.schema.nodes[this.options.defaultType].create()
```

Comment:
Should we check if the node type exists on the schema before? If not I'd keep the old behavior of inserting the defaultType.